R/datasummary_crosstab.R

```diff
@@ -0,0 +1,58 @@
+#' Cross tabulations for categorical variables
+#'
+#' Convenience function to tabulate counts, cell percentages, and row/column
+#' percentages for categorical variables.
+#'
+#' @param formula A two-sided formula to describe the table: rows ~ columns, where
+#' rows and columns are variables in the data. Both rows and column may contain
+#' interactions, e.g., var1 * var2 ~ var3.
+#' @param statistic A formula of the form 1 ~ 1 + N + Percent("row"). The left-hand
+#' side may only be empty or contain a 1 to include row totals.
+#' The right-hand side may contain: 1 for column totals, N for counts, Percent() for
+#' cell percentages, Percent("row") for row percentages, Percent("col") for column
+#' percentages.
+#' @param data A data.frame (or tibble)
+#' @param sparse_header TRUE or FALSE. TRUE eliminates column headers which
+#' have a unique label across all columns, except for the row immediately above
+#' the data. FALSE keeps all headers. The order in which terms are entered in
+#' the formula determines the order in which headers appear.
+#' @param ... passed to datasummary
+#' @return a crosstab
+#' @details
+#' X
+#' @examples
+#' \dontrun{
+#' }
+#'
+#' @export
+datasummary_crosstab <- function(
+        formula,
+        statistic = 1 ~ 1 + N + Percent("row"),
+        data,
+        sparse_header = FALSE, ...) {
```

Comment:
include all relevant arguments explicitly instead of ellipsis.

think about whether `fmt=1` makes sense for percentages (avoid ridiculous precision)

Reply:
fmt is a good one! Do you mean to include others as well?

Reply:
Yeah, they won't appear in the docs. OK, I'll copy some more options.